tests/validation/tests/v3_api/test_workload.py

```diff
@@ -858,3 +858,306 @@ def setup_project_by_role(role, remove_resource):
+                                            containers=con,
+                                            namespaceId=ns.id,
+                                            daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        with pytest.raises(ApiError) as e:
+            p_client.delete(workload)
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'Forbidden'
+        remove_resource(workload)
+        if role == CLUSTER_MEMBER:
+            remove_resource(project)
+
+@if_test_rbac
+@pytest.mark.parametrize("role", [CLUSTER_MEMBER,
+                                  PROJECT_MEMBER, PROJECT_OWNER,
+                                  PROJECT_READ_ONLY, CLUSTER_OWNER])
+def test_rbac_wl_daemonset_delete_negative(remove_resource):
+    if role == CLUSTER_OWNER:
+        # cluster owner can edit daemonsets in any project in the cluster
+        project = rbac_get_project()
+        user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        p2 = rbac_get_unshared_project()
+        p_client = get_project_client_for_token(project, user_token)
+        p_client2 = get_project_client_for_token(p2, user_token)
+        ns2 = rbac_get_unshared_ns()
+        name = random_test_name("default")
+        con = [{"name": "test1",
+                "image": TEST_IMAGE}]
+        workload = p_client2.create_workload(name=name,
+                                             containers=con,
+                                             namespaceId=ns.id,
+                                             daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        con = [{"name": "test1", "image": "nginx"}]
+        p_client.update(workload, containers=con)
+        validate_workload_image(p_client, workload, "nginx", ns)
+        p_client.delete(workload)
+
+    else:
+        project = rbac_get_unshared_project()
+        user_token = rbac_get_user_token_by_role(role)
+        p_client = get_project_client_for_token(project, user_token)
+        # roles cannot delete daemonsets in projects they cannot access
+        # deploy a workload as cluster owner
+        cluster_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        cluster_owner_p_client = get_project_client_for_token(project,
+                                                              cluster_owner_token)
+        ns = rbac_get_namespace()
+        con = [{"name": "test1", "image": TEST_IMAGE}]
+        name = random_test_name("default")
+        workload = cluster_owner_p_client.create_workload(name=name,
+                                                          containers=con,
+                                                          namespaceId=ns.id,
+                                                          daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        # project read-only can NOT delete existing workload
```

Comment:
The comment here is not accurate, in this test case no one except the cluster owner can delete the workload.

Reply:
Removed comment.